react/components/Collapsible.js

```diff
@@ -1,51 +1,64 @@
-import React from 'react'
-import PropTypes from 'prop-types'
+import React, { Fragment } from 'react'
 import classNames from 'classnames'
-import withScrollAnimation from './ScrollAnimation'
-import { Border } from './Helpers'
+import { useSpring, animated, config as springPresets } from 'react-spring'
 import useDevice from '../hooks/useDevice'
+import useScrollDirection from '../hooks/useScrollDirection'
+import Border from './Helpers/Border'
+import { collapsible, lean } from '../defaults'
 
-import header from '../store-header.css'
+import styles from '../store-header.css'
 
-const Collapsible = ({
-  children,
-  leanMode,
-  animation,
-  didAnimate,
-  onAnimate,
-}) => {
+/**
+ * Represents a collapsible part of the header
+ */
+const Collapsible = ({ children, leanMode, collapsibleAnimation }) => {
   const { desktop } = useDevice()
-  onAnimate(didAnimate)
+  const { scroll, scrollingUp } = useScrollDirection()
+  const {
+    onScroll,
+    anchor,
+    always,
+    from,
+    to,
+    preset,
+    config,
+  } = collapsibleAnimation
+
+  let animationTrigger = true
+
+  if (onScroll) {
+    const animateWhen = scroll < anchor
+    const elastic = always ? scrollingUp : false
+    animationTrigger = animateWhen || elastic
+  }
+
+  const props = useSpring({
+    config: preset ? springPresets[preset] : config,
+    transform: animationTrigger ? `translateY(${to}rem)` : `translateY(${-from}rem)`,
```

Comment:
this is nitpicky at this point, but:

```suggestion
    transform: animationTrigger ? `translate3d(0, ${to}rem, 0)` : `translate3d(0, ${-from}rem, 0)`,
```

should have slightly better performance, as it forces GPU acceleration, and it's essentially "free" (the limits it does have are not too relevant in this case)

Reply:
Done ✅